crates/bevy_solari/src/realtime/restir_gi.wgsl

```diff
@@ -116,16 +116,31 @@ fn generate_initial_reservoir(world_position: vec3<f32>, world_normal: vec3<f32>
 }
 
 fn load_temporal_reservoir(pixel_id: vec2<u32>, depth: f32, world_position: vec3<f32>, world_normal: vec3<f32>) -> NeighborInfo {
+    var temporal = NeighborInfo(empty_reservoir(), vec3(0.0), vec3(0.0), vec3(0.0));
+
     let motion_vector = textureLoad(motion_vectors, pixel_id, 0).xy;
     let temporal_pixel_id_float = round(vec2<f32>(pixel_id) - (motion_vector * view.main_pass_viewport.zw));
-    let temporal_pixel_id = permute_pixel(vec2<u32>(temporal_pixel_id_float), constants.frame_index, view.viewport.zw);
 
     // Check if the current pixel was off screen during the previous frame (current pixel is newly visible),
     // or if all temporal history should assumed to be invalid
     if any(temporal_pixel_id_float < vec2(0.0)) || any(temporal_pixel_id_float >= view.main_pass_viewport.zw) || bool(constants.reset) {
-        return NeighborInfo(empty_reservoir(), vec3(0.0), vec3(0.0), vec3(0.0));
+        return temporal;
```

Comment:
Same here (maybe the empty `NeighborInfo` construction could go into its own function too).